flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/resources/FlinkPod.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.kubernetes.kubeclient.resources;
+
+import org.apache.flink.configuration.Configuration;
+
+import io.fabric8.kubernetes.api.model.ContainerStatus;
+import io.fabric8.kubernetes.api.model.Pod;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Represent FlinkPod resource in kubernetes.
+ */
+public class FlinkPod extends Resource<Pod> {
+
+	public FlinkPod(Configuration flinkConfig) {
+		this(flinkConfig, new Pod());
+	}
+
+	public FlinkPod(Configuration flinkConfig, Pod pod) {
+		super(flinkConfig, pod);
+	}
+
+	public String getName() {
+		return this.getInternalResource().getMetadata().getName();
+	}
+
+	public boolean isTerminated() {
+		if (getInternalResource().getStatus() != null &&
+			!getInternalResource().getStatus().getContainerStatuses().isEmpty()) {
+
+			List<ContainerStatus> terminatedStates =
+				getInternalResource().getStatus().getContainerStatuses().stream()
+					.filter(e -> e.getState() != null && e.getState().getTerminated() != null).collect(Collectors.toList());
+			return !terminatedStates.isEmpty();
+		}
```

Comment:
The block can be simpler and maybe faster with
```
		if (internalResource.getStatus() != null) {
			return internalResource.getStatus().getContainerStatuses().stream()
				.filter(e -> e.getState() != null && e.getState().getTerminated() != null)
				.findAny()
				.isPresent();
		}
```

Reply:
Nice. I will use `.anyMatch(e -> e.getState() != null && e.getState().getTerminated() != null)` instead.